fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/SuspendingRequest.kt

```diff
@@ -10,7 +10,7 @@ import java.io.InterruptedIOException
 class SuspendingRequest(private val request: Request) {
     var interruptCallback: ((Request) -> Unit)? = null
 
-    suspend fun awaitResult(): Result<Response, FuelError> {
+    suspend fun awaitResult(): Result<Response, Exception> {
```

Comment:
I think it might be a better idea to unify Exception in Fuel as `FuelError` as we can provide more information to the user what went wrong .... Shall we just catch the exception and wrap it as `FuelError`?

Reply:
thats what this is doing... this is collecting the errors and later in the pipeline .mapErrors is executed

i just matched the synchronous pipeline and it happened to work

Reply:
i thin kresult managest the exception catching and the problem here seems to be that in one case mapErrors is assuming a different type and casts exceptions

also we seem to be doing mapErrors in multiple places directly after another in the pipline